src/tunable_static_tf_broadcaster.cpp

```diff
@@ -0,0 +1,138 @@
+#include "tunable_static_tf_broadcaster.h"
+#include "tunable_static_tf_broadcaster/TfConfig.h"
+#include <tf2/LinearMath/Quaternion.h>
+
+namespace tunable_static_tf_broadcaster
+{
+
+TunableStaticTfBroadcaster::TunableStaticTfBroadcaster()
+ : private_nh_("~")
+{
+    // Initialize transform
+    transform_.transform.translation.x = 0.0;
+    transform_.transform.translation.y = 0.0;
+    transform_.transform.translation.z = 0.0;
+    tf2::Quaternion quaternion;
+    quaternion.setRPY(0.0, 0.0, 0.0);
+    transform_.transform.rotation.x = quaternion.x();
+    transform_.transform.rotation.y = quaternion.y();
+    transform_.transform.rotation.z = quaternion.z();
+    transform_.transform.rotation.w = quaternion.w();
+
+    // Get configuration from param
+    publish_rate_ = private_nh_.param<int>("rate", 10);
+    transform_.header.frame_id = private_nh_.param<std::string>("header_frame", "world");
+    transform_.child_frame_id = private_nh_.param<std::string>("child_frame", "base_link");
+
+    ROS_DEBUG("[tunable_static_tf_broadcaster] Setting parameters... {rate: %d, header_frame: %s, child_frame: %s}",
+        publish_rate_, transform_.header.frame_id.c_str(), transform_.child_frame_id.c_str());
+
+    // Get if static param is set
+    if( private_nh_.hasParam("tf_x")
+     || private_nh_.hasParam("tf_y")
+     || private_nh_.hasParam("tf_z")
+     || private_nh_.hasParam("tf_roll")
+     || private_nh_.hasParam("tf_pitch")
+     || private_nh_.hasParam("tf_yaw") )
+    {
+        has_initial_static_params_ = true;
+    }
+    else
+    {
+        has_initial_static_params_ = false;
+    }
+
+    // Setup dynamic reconfigure
+    dynamic_reconfigure_server_ = std::make_shared<dynamic_reconfigure::Server<TfConfig> >();
+    dynamic_reconfigure::Server<TfConfig>::CallbackType f;
+    f = boost::bind<void>(&TunableStaticTfBroadcaster::dynamicReconfigureCallback, this, _1, _2);
```

Comment:
In ubuntu16.04, ros kinetic environment, an error occurred when building as it is.
Please remove <void> if there is no problem in other environments.

Reply:
I fixed that.